vignettes/adtr.Rmd

```diff
@@ -0,0 +1,725 @@
+    mode = "first",
+    check_type = "none"
+  )
+```
+
+```{r, echo=FALSE}
+dataset_vignette(
+  adtr_sum %>%
+    arrange(USUBJID, AVISITN) %>%
+    select(USUBJID, PARAMCD, AVISIT, ADY, AVAL, NADIR, everything()),
+  display_vars = exprs(USUBJID, PARAMCD, AVISIT, ADY, AVAL, NADIR)
+)
+```  
+
+## Derive Change from Baseline (`CHG`, `PCHG`) {#chg}
+
+These functions are available in 'admiral', below examples show its usage for
+`ADTR`.  We calculate the change from baseline (`CHG`, `CHGNAD`) and percentage change from baseline (`PCHG` , `PCHGNAD`) for the parameter `SDIAM` (using `AVAL` - `BASE`) the variable `NADIR` (using `AVAL` - `NADIR`).
+
+```{r}
+adtr_sum <- adtr_sum %>%
+  derive_var_chg() %>%
+  derive_var_pchg() %>%
+  mutate(
+    CHGNAD = AVAL - NADIR,
+    PCHGNAD = if_else(NADIR == 0, NA_real_, 100 * CHGNAD / NADIR)
+  )
+```
+
+```{r, echo=FALSE}
+dataset_vignette(
+  arrange(adtr_sum, USUBJID, AVISITN),
+  display_vars = exprs(USUBJID, AVISIT, AVAL, NADIR, BASE, CHG, PCHG, CHGNAD, PCHGNAD)
+)
+```  
+
+## Derive Additional Flag Variables {#addflag}
+
+Depending on the analysis flags you require, you may want to create additional
+flag variables that will then aid setting up any additional analysis flag
+variables defined as per your company specifications.
+
+### Derive `PDFL`
+
+In this example, we want to flag when a patient has had `PD`.  This could be done in a number of ways:
+
+1. Take the date of `PD` from `ADRS`
+
+    ```{r eval=TRUE}
+adtr_sum <- adtr_sum %>%
+  derive_var_merged_exist_flag(
+    dataset_add = adrs,
+    filter_add = PARAMCD == "PD",
+    by_vars = exprs(USUBJID, ADT),
+    new_var = PDFL,
+    condition = AVALC == "Y"
+  )
+    ```
+
+    ```{r, echo=FALSE}
+dataset_vignette(
+  arrange(adtr_sum, USUBJID, AVISITN),
+  display_vars = exprs(USUBJID, AVISIT, AVAL, NADIR, CHGNAD, PCHGNAD, PDFL)
+)
+
+adtr_sum <- select(adtr_sum, -PDFL)
+    ```
+
+2. Take the first date when the overall response is `PD` from `RS` 
+
+    ```{r eval=TRUE}
+adtr_sum <- adtr_sum %>%
+  derive_var_merged_exist_flag(
+    dataset_add = derive_vars_dt(
+      rs,
+      dtc = RSDTC,
+      new_vars_prefix = "A",
+      highest_imputation = "D",
+      flag_imputation = "none"
+    ),
+    filter_add = RSTESTCD == "OVRLRESP" & RSEVAL == "INVESTIGATOR",
+    by_vars = exprs(USUBJID, ADT),
+    new_var = PDFL,
+    condition = RSSTRESC == "PD"
+  )
+    ```
+
+    ```{r, echo=FALSE}
+dataset_vignette(
+  arrange(adtr_sum, USUBJID, AVISITN),
+  display_vars = exprs(USUBJID, AVISIT, AVAL, NADIR, CHGNAD, PCHGNAD, PDFL)
+)
+
+adtr_sum <- select(adtr_sum, -PDFL)
+    ```
+
+3. Target lesion response of no `CR` at the current observation visit and but a `CR` response at the `NADIR` observation
+
+    ```{r}
+adtr_sum <- adtr_sum %>% mutate(
+  CRFL = if_else(AVAL == 0 & ANL01FL == "Y", "Y", NA_character_),
+  CRNFL = if_else(NADIR == 0, "Y", NA_character_),
+  PDFL = if_else(is.na(CRFL) & CRNFL == "Y", "Y", NA_character_)
```

Comment:
why we are again creating PDFL here, when we already provided example derviation for PDFL above

Reply:
For `PDFL` we show four options how it could be derived. The users would need to select one of them.